src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -26,17 +33,157 @@ const ActorInit Shot_Sun_InitVars = {
+        this->actionFunc = ShotSun_HyliaSunUpdate;
+        this->actor.flags &= ~1;
+    }
+}
+
+void ShotSun_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ShotSun* this = THIS;
+    s32 params = this->actor.params & 0xFF;
+
+    if (params != 0x40 && params != 0x41) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void ShotSun_SpawnFairy(ShotSun* this, GlobalContext* globalCtx) {
+    s32 params = this->actor.params & 0xFF;
+    s32 fairyParams;
+
+    if (this->timer > 0) {
+        this->timer--;
+        return;
+    }
+
+    switch (params) {
+        case 0x40:
+            fairyParams = 7;
+            break;
+        case 0x41:
+            fairyParams = 7;
+            break;
+    }
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, this->actor.initPosRot.pos.x,
+                this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, fairyParams);
+
+    Actor_Kill(&this->actor);
+}
+
+void ShotSun_TriggerFairy(ShotSun* this, GlobalContext* globalCtx) {
+    if ((func_8005B198() == this->actor.type) || (this->timer != 0)) {
+        this->actionFunc = ShotSun_SpawnFairy;
+        this->timer = 50;
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_DEMO_KANKYO, this->actor.initPosRot.pos.x,
+                    this->actor.initPosRot.pos.y, this->actor.initPosRot.pos.z, 0, 0, 0, 0x11);
+
+        func_80078914(&this->actor.projectedPos, NA_SE_EV_TRE_BOX_APPEAR);
+    }
+}
+
+void func_80BADF0C(ShotSun* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 pad;
+    s32 params = this->actor.params & 0xFF;
+
+    if (func_800CB650(&this->actor.posRot.pos, &player->actor.posRot.pos) > 22500.0f) {
+        this->unk_1A4[0] = 0;
+    } else {
+        if (this->unk_1A4[0] == 0) {
+            if (!(player->stateFlags2 & 0x1000000)) {
+                player->stateFlags2 |= 0x800000;
+                return;
+            } else {
+                this->unk_1A4[0] = 1;
+            }
+        }
+        if (this->unk_1A4[0] == 1) {
+            func_8010BD58(globalCtx, 1U);
+            this->unk_1A4[0] = 2;
+            return;
+        }
+
+        if (this->unk_1A4[0] == 2 && globalCtx->msgCtx.unk_E3EE == 4) {
+            if ((params == 0x40 && globalCtx->msgCtx.unk_E3EC == 9) ||
+                (params == 0x41 && globalCtx->msgCtx.unk_E3EC == 0xB)) {
+                this->actionFunc = ShotSun_TriggerFairy;
+                func_80080480(globalCtx, this);
+                this->timer = 0;
+            } else {
+                this->unk_1A4[0] = 0;
+            }
+            this->unk_1A4[0] = 0;
+        }
+    }
+}
+
+void ShotSun_HyliaSunUpdate(ShotSun* this, GlobalContext* globalCtx) {
+    Vec3s cylinderPos;
+    Player* player = PLAYER;
+    EnItem00* collectible;
+    s32 pad;
+    Vec3f spawnPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Shot_Sun/func_80BADDCC.s")
+    if ((this->collider.base.acFlags & 2) != 0) {
```

Comment:
oh i must have missed this last time, this a flags cehck so you dont need != 0

Reply:
Right, should I also change the if condition in ShotSun_TriggerFairy to be
`func_8005B198() == this->actor.type || this->timer`?

Reply:
only for boolean values. in that case its actually checking the the timer to not be 0

Reply:
Got it.